src/main/java/eu/europa/ted/efx/xpath/XPathScriptGenerator.java

```diff
@@ -479,8 +480,23 @@ public StringExpression getStringLiteralFromUnquotedString(String value) {
     return new StringExpression("'" + value + "'", true);
   }
 
+  @Override
+  public StringExpression getPreferredLanguage(PathExpression fieldReference) {
+    final String variableName = "$__LANG__";
+      String languages = "('" + String.join("','", this.translatorOptions.getAllLanguage3LetterCodes()) + "')";
```

Comment:
I would expect this to use the "ted:preferred-languages" function, as it is done in `XPathScriptGeneratorV1.java`
Here it just uses all language codes in the order given by TranslatorOptions.

Reply:
Indeed. Fixed.